osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+# Set up django configuration
+import django
+from kobo.client.constants import FINISHED_STATES
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+django.setup()
+from kobo.hub.models import Task  # noqa
+
+from osh.hub.scan.models import RetentionPolicySetting  # noqa: E402
+from osh.hub.scan.models import TaskResultsRemoval  # noqa: E402
+
+logger = logging.getLogger("osh.hub.scripts.osh-retention")
+
+
+DEFAULT_RETENTION_DAYS = 30
+
+
+def delete_tasks_results(tasks_to_delete):
+    for task, retention_policy_setting in tasks_to_delete:
+        try:
+            logger.debug('The following task results will be deleted: %s.', task.id)
+            path_to_delete = Task.get_task_dir(task.id)
+            shutil.rmtree(path_to_delete)
+            logging.debug('The following folder was deleted: %s', path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+            return
```

Comment:
`return` -> `continue`

If the data is for whatever reason deleted from the file system already, it is not a reason to entirely give up on the retention policy a skip removing the other 200 000 tasks that should be deleted.

Reply:
Actually, we need to distinguish two cases:

1. the data is already deleted - then we should mark as deleted in the database
2. the data still exists but cannot be deleted - then we should try it next time again

Reply:
Modified